go/data-source-sql/countries/countries.go

```diff
@@ -0,0 +1,81 @@
+package countries
+
+// Access to a database of country names.
+
+import (
+	"fmt"
+	"log"
+	"database/sql"
+	"strings"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/signavio/connector-examples/go/data-source-sql/configuration"
+)
+
+var databaseDriver = "mysql"
+var databaseUrl = "cldr:cldr@tcp(127.0.0.1:3306)/cldr"
+
+type Countries struct {
+  db *sql.DB
+}
+
+// JSON type required by the connector API.
+type Country struct {
+	Code string `json:"id"`
+	Name string `json:"name"`
+}
+
+// Connect to the database.
+func NewCountries(configuration configuration.Configuration) (*Countries) {
+  countries := Countries{}
+  countries.db = Database(configuration)
+  return &countries
+}
+
+// Connects to the database and tests the connection
+func Database(configuration configuration.Configuration) *sql.DB {
+	db, err := sql.Open(configuration.DatabaseDriver, configuration.DatabaseUrl)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	if err = db.Ping(); err != nil {
+		log.Fatal(err.Error())
+	}
+	log.Print("Connected to database")
+	return db
+}
+
+// Finds a country by code.
```

Comment:
I think its more idiomatic to have the `FindOne` function return an `error` as its second argument (i.e `FindOne(code string) (Country, error)`). You can always log the error from the calling function. The same applies for the `Find` function.

Reply:
That's exactly what I need! I don't even want a typed `nil` because I need a value for the missing country code.

Reply:
@sdaros I’ve added the new error type. Is [this](https://github.com/signavio/connector-examples/blob/affae2c172b92ed7bf7db3414148f73bd7d5179a/go/data-source-sql/countries/countries.go#L60L62) a good way to make `FindOne` return the new type, given that there could be various SQL errors?

Reply:
I think the preferred way would be to check the error type using a [type insertion](https://golang.org/doc/effective_go.html#interface_conversions) instead of matching against the error message. What you would do in this case is check the documentation for the [sql](http://devdocs.io/go/database/sql/index#pkg-variables) package to see if it exposes a custom error type. See my other comment for what this would look like